### ISSUE TITLE
outdated sentence in perlsec

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#16571]
Migrated from [rt.perl.org#133230](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=133230) (status was 'open')

Searchable as RT133230$

**Comment by p5pRT** [comment]
### From lortav72@gmail.com

This is a bug report for perl from lortav72@&#8203;gmail\.com\,
generated with the help of perlbug 1\.39 running under perl 5\.14\.2\.

In perlsec\, the 'Taint mode and @&#8203;INC' paragraph begins with the sentence&#8203;:
'When the taint mode \(\-T \) is in effect\, the "\." directory is removed from
@&#8203;INC \,\.\.'
which it seems no more needed because "\." was removed from @&#8203;INC since
5\.26

Possible better sentence can be&#8203;:

For perl version prior of 5\.26 hen the taint mode \(\-T \) is in effect\, the
"\." directory is removed from @&#8203;INC\, and
the environment variables PERL5LIB and PERLLIB are ignored by Perl \(note
that since Perl 5\.26 "\." is no more present in
@&#8203;INC anyway\)\.

thanks for reading

lortav72@&#8203;gmail\.com\-\-\-
Flags&#8203;:
&nbsp;    category=docs
&nbsp;    severity=low


---
Site configuration information for perl 5\.14\.2&#8203;:

Configured by 1 at Tue Nov 22 18&#8203;:26&#8203;:53 2011\.

Summary of my perl5 \(revision 5 version 14 subversion 2\) configuration&#8203;:

&nbsp;  Platform&#8203;:
&nbsp;    osname=MSWin32\, osvers=4\.0\, archname=MSWin32\-x86\-multi\-thread
&nbsp;    uname='Win32 strawberryperl 5\.14\.2\.1\-portable \#1 Tue Nov 22 18&#8203;:24&#8203;:29
2011 i386'
&nbsp;    config\_args='undef'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=undef
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, ccflags =' \-s \-O2 \-DWIN32  \-DPERL\_TEXTMODE\_SCRIPTS
\-DUSE\_SITECUSTOMIZE \-DPERL\_IMPLICIT\_CONTEXT \-DPERL\_IMPL
ICIT\_SYS \-fno\-strict\-aliasing \-mms\-bitfields'\,
&nbsp;    optimize='\-s \-O2'\,
&nbsp;    cppflags='\-DWIN32'
&nbsp;    ccversion=''\, gccversion='4\.4\.7'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=undef\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='long long'\,
lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='g\+\+\.exe'\, ldflags ='\-s \-L"C&#8203;:\\ulisse\\strawberry\\perl\\lib\\CORE"
\-L"C&#8203;:\\ulisse\\strawberry\\c\\lib"'
&nbsp;    libpth=C&#8203;:\\ulisse\\strawberry\\c\\lib
C&#8203;:\\ulisse\\strawberry\\c\\i686\-w64\-mingw32\\lib
&nbsp;    libs=\-lmoldname \-lkernel32 \-luser32 \-lgdi32 \-lwinspool \-lcomdlg32
\-ladvapi32 \-lshell32 \-lole32 \-loleaut32 \-lnetapi32
 \-luuid \-lws2\_32 \-lmpr \-lwinmm \-lversion \-lodbc32 \-lodbccp32 \-lcomctl32
&nbsp;    perllibs=\-lmoldname \-lkernel32 \-luser32 \-lgdi32 \-lwinspool \-lcomdlg32
\-ladvapi32 \-lshell32 \-lole32 \-loleaut32 \-lneta
pi32 \-luuid \-lws2\_32 \-lmpr \-lwinmm \-lversion \-lodbc32 \-lodbccp32 \-lcomctl32
&nbsp;    libc=\, so=dll\, useshrplib=true\, libperl=libperl514\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_win32\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags='\-mdll \-s
\-L"C&#8203;:\\ulisse\\strawberry\\perl\\lib\\CORE" \-L"C&#8203;:\\ulisse\\strawberry\\c\\lib"'

Locally applied patches&#8203;:




---
@&#8203;INC for perl 5\.14\.2&#8203;:
&nbsp;    C&#8203;:/ulisse/strawberry/perl/site/lib/MSWin32\-x86\-multi\-thread
&nbsp;    C&#8203;:/ulisse/strawberry/perl/site/lib
&nbsp;    C&#8203;:/ulisse/strawberry/perl/vendor/lib
&nbsp;    C&#8203;:/ulisse/strawberry/perl/lib
&nbsp;    \.



---
Environment for perl 5\.14\.2&#8203;:
&nbsp;    HOME \(unset\)
&nbsp;    LANG \(unset\)
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)

PATH=C&#8203;:\\ulisse\\strawberry\\perl\\site\\bin;C&#8203;:\\ulisse\\strawberry\\perl\\bin;C&#8203;:\\ulisse\\strawberry\\c\\bin;C&#8203;:\\ulisse\\bin\\UnxUt
ils\\usr\\local\\wbin;C&#8203;:\\Windows;C&#8203;:\\Windows\\system32;;\.;
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    PERL\_JSON\_BACKEND=JSON&#8203;::XS
&nbsp;    PERL\_RL=Perl
&nbsp;    PERL\_YAML\_BACKEND=YAML
&nbsp;    SHELL \(unset\)

**Comment by p5pRT** [comment]
### From @tonycoz

On Mon\, 28 May 2018 13&#8203;:41&#8203;:43 \-0700\, lortav72@&#8203;gmail\.com wrote&#8203;:
> In perlsec\, the 'Taint mode and @&#8203;INC' paragraph begins with the
> sentence&#8203;:
> 'When the taint mode \(\-T \) is in effect\, the "\." directory is removed
> from
> @&#8203;INC \,\.\.'
> which it seems no more needed because "\." was removed from @&#8203;INC since
> 5\.26
> 
> Possible better sentence can be&#8203;:
> 
> For perl version prior of 5\.26 hen the taint mode \(\-T \) is in effect\,
> the
> "\." directory is removed from @&#8203;INC\, and
> the environment variables PERL5LIB and PERLLIB are ignored by Perl
> \(note
> that since Perl 5\.26 "\." is no more present in
> @&#8203;INC anyway\)\.
> 

Something like the attached\, maybe\.

Tony

**Comment by p5pRT** [comment]
### From @tonycoz

<details><summary>0001-perl-133230-.-is-no-longer-in-INC-by-default.patch</summary>

```diff
From fd1d01e82034d5a4e1bde1a7db296bb10c1ac479 Mon Sep 17 00:00:00 2001
From: Tony Cook <tony@develop-help.com>
Date: Thu, 31 May 2018 11:52:45 +1000
Subject: (perl #133230) . is no longer in @INC by default

---
 pod/perlmodlib.PL | 3 ++-
 pod/perlsec.pod   | 6 +++---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/pod/perlmodlib.PL b/pod/perlmodlib.PL
index b92f9ca151..0af82929e5 100644
--- a/pod/perlmodlib.PL
+++ b/pod/perlmodlib.PL
@@ -186,7 +186,8 @@ double quotes should be used instead of single quotes).
       'find { wanted => sub { print canonpath $_ if /\.pm\z/ },
       no_chdir => 1 }, @INC'
 
-(The -T is here to prevent '.' from being listed in @INC.)
+(The -T is here to prevent @INC from being populated by C<PERL5LIB>,
+C<PERL5LIB> and C<PERL_USE_UNSAFE_INC>.)
 They should all have their own documentation installed and accessible
 via your system man(1) command.  If you do not have a B<find>
 program, you can use the Perl B<find2perl> program instead, which
diff --git a/pod/perlsec.pod b/pod/perlsec.pod
index b210445685..9bc40f216b 100644
--- a/pod/perlsec.pod
+++ b/pod/perlsec.pod
@@ -245,11 +245,11 @@ Unix-like environments that support #! and setuid or setgid scripts.)
 
 =head2 Taint mode and @INC
 
-When the taint mode (C<-T>) is in effect, the "." directory is removed
-from C<@INC>, and the environment variables C<PERL5LIB> and C<PERLLIB>
+When the taint mode (C<-T>) is in effect, the environment variables
+C<PERL5LIB>, C<PERLLIB> and C<PERL_USE_UNSAFE_INC>
 are ignored by Perl.  You can still adjust C<@INC> from outside the
 program by using the C<-I> command line option as explained in
-L<perlrun>.  The two environment variables are ignored because
+L<perlrun>.  The three environment variables are ignored because
 they are obscured, and a user running a program could be unaware that
 they are set, whereas the C<-I> option is clearly visible and
 therefore permitted.
-- 
2.11.0


```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @hvds

On Wed\, 30 May 2018 18&#8203;:53&#8203;:40 \-0700\, tonyc wrote&#8203;:
>\+\(The \-T is here to prevent @&#8203;INC from being populated by C\<PERL5LIB>\, 
>\+C\<PERL5LIB> and C\<PERL\_USE\_UNSAFE\_INC>\.\)

s/PERL5LIB/PERLLIB/ the second time\, I assume\.

**Comment by khwilliamson** [comment]
Applied with @hvds fix as 96acb8b6c1fe2f0efbfb5d45e3efad4838fa411b
